### ISSUE TITLE
sql integrity error at DFK shutdown after failed tasks

### ISSUE THREAD

**Issue by benclifford** [Parsl/parsl#1100]
I'm getting the below exception from monitoring code at DFK shutdown. My gut feeling is it is coming from trying to record tasks that have not been submitted for execution (because they are waiting on dependencies) and so don't have a submit time - but I haven't looking in the code to confirm this.

```
parsl.dataflow.error.DependencyError: [3] Dependency failure from: [[2] Dependency failure from: [AppFailure('[import_ci_hsc] App failed with exit code: 128', 128)]]
2019-06-27 14:41:01 parsl.dataflow.dflow:822 [INFO]  DFK cleanup initiated
INFO:parsl.dataflow.dflow:DFK cleanup initiated
2019-06-27 14:41:01 parsl.dataflow.dflow:733 [INFO]  Summary of tasks in DFK:
INFO:parsl.dataflow.dflow:Summary of tasks in DFK:
2019-06-27 14:41:01 parsl.dataflow.dflow:764 [INFO]  Tasks in state States.done: 1, 5
INFO:parsl.dataflow.dflow:Tasks in state States.done: 1, 5
2019-06-27 14:41:01 parsl.dataflow.dflow:764 [INFO]  Tasks in state States.failed: 0, 4
INFO:parsl.dataflow.dflow:Tasks in state States.failed: 0, 4
2019-06-27 14:41:01 parsl.dataflow.dflow:764 [INFO]  Tasks in state States.dep_fail: 2, 3
INFO:parsl.dataflow.dflow:Tasks in state States.dep_fail: 2, 3
2019-06-27 14:41:01 parsl.dataflow.dflow:771 [INFO]  End of summary
INFO:parsl.dataflow.dflow:End of summary
2019-06-27 14:41:01 parsl.dataflow.dflow:846 [INFO]  Terminating flow_control and strategy threads
INFO:parsl.dataflow.dflow:Terminating flow_control and strategy threads
2019-06-27 14:41:01 parsl.executors.threads:96 [DEBUG]  Done with executor shutdown
DEBUG:parsl.executors.threads:Done with executor shutdown
2019-06-27 14:41:01 parsl.executors.threads:96 [DEBUG]  Done with executor shutdown
DEBUG:parsl.executors.threads:Done with executor shutdown
2019-06-27 14:41:01 parsl.data_provider.data_manager:98 [DEBUG]  Done with executor shutdown
DEBUG:parsl.data_provider.data_manager:Done with executor shutdown
INFO:monitoring_hub:Terminating Monitoring Hub
INFO:monitoring_hub:Waiting Hub to receive all messages and terminate
Process Process-2:
Traceback (most recent call last):
  File "/opt/lsst/software/stack/stack/miniconda3-4.5.4-fcd27eb/Linux64/sqlalchemy/1.2.16/lib/python/SQLAlchemy-1.2.16-py3.6-linux-x86_64.egg/sqlalchemy/engine/base.py", line 1210, in _execute_context
    cursor, statement, parameters, context
  File "/opt/lsst/software/stack/stack/miniconda3-4.5.4-fcd27eb/Linux64/sqlalchemy/1.2.16/lib/python/SQLAlchemy-1.2.16-py3.6-linux-x86_64.egg/sqlalchemy/engine/default.py", line 533, in do_executemany
    cursor.executemany(statement, parameters)
sqlite3.IntegrityError: NOT NULL constraint failed: task.task_time_submitted
```

**Comment by ZhuozhaoLi** [comment]
Do you have a script to reproduce this error?

**Comment by benclifford** [comment]
I'll add a test case

**Comment by ZhuozhaoLi** [comment]
I think this is caused by these two lines: https://github.com/Parsl/parsl/blob/f98fdd076b9bb0edb5c9d99fb116de933d777a87/parsl/dataflow/dflow.py#L261-L262

As the tasks failed, so they do not have any `submitted_time`

**Comment by benclifford** [comment]
#1111 sometimes shows this (athough it never fails in the pytest sense)

**Comment by benclifford** [comment]
This should be fixed by #1109
